### PR TITLE
Design System: Add loading animation

### LINKED ISSUE #6622: Dashboard: Loading State
## Feature Description

This has been brought up a few times in conversation, I might be duplicating a ticket that I can't find. We need a loading state on the dashboard to prevent flashing "empty" content. This will also help decrease perceived time to interaction. 

We need design guidance to implement this. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

Dashboard shows loading state while loading.

### PR DESCRIPTION
**Pull request by samwhale**
## Context

Dashboard needs a loading animation to indicate that there is data loading.
[Designs](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/?node-id=6113%3A471417)
[Animation Reference (3rd on the right)](https://dribbble.com/shots/7330802-Loading-Spinner/attachments/247146?mode=media)

## Summary

Adds a loading animation the design system and the dashboard.

## Relevant Technical Choices

- svg coming from figma was weird so I implemented the spinner with react components and css animations

## To-do

none

## User-facing changes

none

## Testing Instructions

1. Go to Storybook -> Design System -> LoadingSpinner
2. Play with the knobs
3. Should read a status if using a screen reader/screen reading software

### QA

- [x] This is a non-user-facing change and requires no QA

### UAT

- [x] UAT should use the same steps as above.

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses #6622

### CODE REVIEW

**Comment by samwhale** [comment]
Once this is merged we will add the loading spinner to the dashboard. Split up the prs because this was getting too large

**Review by BrittanyIRL — APPROVED** [review]
this is awesome!

File: assets/src/design-system/components/loadingSpinner/index.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    }
+    ${`${(circleIndex * 100) / numCircles}%`} {
+      opacity: 1
+    }
+    ${`${((circleIndex + 1) * 100) / numCircles}%`} {
+      opacity: 0.3
+    }
+    ${`${((circleIndex + 1) * 100) / numCircles + 1}%`} {
+      opacity: 1
+    }
```
Comment (maxyinger) [inline assets/src/design-system/components/loadingSpinner/index.js:81]: ooo v-nice

**Review by maxyinger — APPROVED** [review]
Beautiful 👏

**Review by embarks — APPROVED** [review]
Nice! I like how you made the visuals adjustable.

**Pull request merged**